tools/lz4-compress.js

```diff
@@ -1,157 +1,60 @@
+#!/usr/bin/env node
 // Copyright 2015 The Emscripten Authors.  All rights reserved.
 // Emscripten is available under two separate licenses, the MIT license and the
 // University of Illinois/NCSA Open Source License.  Both these licenses can be
 // found in the LICENSE file.
 
-// *** Environment setup code ***
-var arguments_ = [];
-var debug = false;
+const fs = require('fs');
+const path = require('path');
 
-var ENVIRONMENT_IS_NODE = typeof process === 'object';
-var ENVIRONMENT_IS_WEB = typeof window === 'object';
-var ENVIRONMENT_IS_WORKER = typeof importScripts === 'function';
-var ENVIRONMENT_IS_SHELL = !ENVIRONMENT_IS_WEB && !ENVIRONMENT_IS_NODE && !ENVIRONMENT_IS_WORKER;
+const arguments_ = process['argv'].slice(2);
+const debug = false;
 
-if (ENVIRONMENT_IS_NODE) {
```

Comment:
These are not necessary anymore? It seems we use these variables in other js files..?

Reply:
Yeah this script is only ever run under node these days.. and its standalone.  This was just copyied and pasted from another script but not needed here.

Reply:
Where do we guarantee this only runs on node? I'm not very familiar with this script's usage, but searching `lz4-compress` in the codebase only turns up `tools/file_packager.py`, which looks it can run elsewhere too... Also do we not run this on d8 as well?

Anyway, if that's the case, does it give a readable error message when someone tries to run this on other environments?

Reply:
This is a completely internal tool just like `src/compiler.js` and `tools/preprocessor.js`.   These tool may once have worked in non-node environments but today we assume they all run under node only.   The code here is completely internal to emscripten and there is exactly one call site.

All of these internal JS scripts require node these days and there is no reason anyone should try to run them under d8 (at least not that I know of).   

If a d8 user did try to run this script and saw `require() is not a function` is should be very obvious that they are trying to run a node script outside of node.